x/core/keeper/grpc_query.go

```diff
@@ -127,3 +130,63 @@ func (q Querier) DataRequestConfig(c context.Context, _ *types.QueryDataRequestC
 	}
 	return &types.QueryDataRequestConfigResponse{DataRequestConfig: dataRequestConfig}, nil
 }
+
+func (q Querier) AccountSeq(c context.Context, req *types.QueryAccountSeqRequest) (*types.QueryAccountSeqResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	staker, err := q.GetStaker(ctx, req.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryAccountSeqResponse{
+		AccountSeq: staker.SequenceNum,
+	}, nil
+}
+
+func (q Querier) IsStakerExecutor(c context.Context, req *types.QueryIsStakerExecutorRequest) (*types.QueryIsStakerExecutorResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	isExecutor, err := q.Keeper.IsStakerExecutor(ctx, req.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryIsStakerExecutorResponse{
+		IsStakerExecutor: isExecutor,
+	}, nil
+}
+
+func (q Querier) IsExecutorEligible(c context.Context, req *types.QueryIsExecutorEligibleRequest) (*types.QueryIsExecutorEligibleResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	// verify the proof
+	hash := req.MsgHash()
+	publicKeyBytes, drIDBytes, proof, err := req.Parts()
```

Comment:
Would these be bytes or hexes?

Reply:
Hmmm might they do base64 encode it though- when we add the tests in a follow up pr we can double check?